File: ts/wam/3934.ts
export const WamWebcNativeUpsellCta = 3934 as const

export interface WebcNativeUpsellCtaEvent {
  /** field 1, wire `webc_native_upsell_cta_source` */
  webcNativeUpsellCtaSource?: WEBCNATIVEUPSELLCTASOURCETYPE
  /** field 2, wire `webc_native_upsell_cta_event_type` */
  webcNativeUpsellCtaEventType?: WEBCNATIVEUPSELLCTAEVENTTYPE
  /** field 3, wire `webc_native_upsell_cta_qr_screen_experiment_group` */
  webcNativeUpsellCtaQrScreenExperimentGroup?: WEBCNATIVEUPSELLCTAQRSCREENEXPERIMENTGROUP
  /** field 4, wire `webc_native_upsell_cta_release_channel` */
  webcNativeUpsellCtaReleaseChannel?: WEBCNATIVEUPSELLCTARELEASECHANNEL
  /** field 5, wire `webc_native_upsell_cta_is_beta_user` */
  webcNativeUpsellCtaIsBetaUser?: boolean
}

File: rust/wam/3934.rs
pub const WAM_WEBC_NATIVE_UPSELL_CTA: u32 = 3934;

#[derive(Debug, Default)]
pub struct WebcNativeUpsellCtaEvent {
    /// field 1, wire `webc_native_upsell_cta_source`
    pub webc_native_upsell_cta_source: Option<WEBCNATIVEUPSELLCTASOURCETYPE>,
    /// field 2, wire `webc_native_upsell_cta_event_type`
    pub webc_native_upsell_cta_event_type: Option<WEBCNATIVEUPSELLCTAEVENTTYPE>,
    /// field 3, wire `webc_native_upsell_cta_qr_screen_experiment_group`
    pub webc_native_upsell_cta_qr_screen_experiment_group: Option<WEBCNATIVEUPSELLCTAQRSCREENEXPERIMENTGROUP>,
    /// field 4, wire `webc_native_upsell_cta_release_channel`
    pub webc_native_upsell_cta_release_channel: Option<WEBCNATIVEUPSELLCTARELEASECHANNEL>,
    /// field 5, wire `webc_native_upsell_cta_is_beta_user`
    pub webc_native_upsell_cta_is_beta_user: Option<bool>,
}